firestore/firestore_snippets/save.go

```diff
@@ -164,23 +174,24 @@ func updateDocNested(ctx context.Context, client *firestore.Client) error {
 	}
 
 	// [START_EXCLUDE]
-	_, err := client.Collection("users").Doc("frank").Set(ctx, initialData)
-	if err != nil {
-		return err
+	_, preErr := client.Collection("users").Doc("frank").Set(ctx, initialData)
```

Comment:
Any particular reason to change this?

Reply:
Yes - this section is excluded in the snippets. Declaring `err :=` down below requires the part in scope of the snippets to reassign rather than declare, which is a compilation error if the snippet is copy and pasted.

Reply:
How about:

```
if _, err := client.Collection("users").Doc("frank").Set(ctx, initialData); err != nil {
  return err
}
```